docs/modules/SPEC.md

```diff
@@ -0,0 +1,80 @@
+# Module Specification
+
+This document attempts to outline the recommended structure of Cosmos SDK modules.
+However, the ideas outlined here are meant to be applied as suggestions. Application
+developers are encouraged to improve upon and contribute to module structure and
+development design.
+
+## Structure
+
+A typical Cosmos SDK module can be structured as follows:
+
+```shell
+x/{module}
+├── abci.go
+├── alias.go
+├── client
+│   ├── cli
+│   │   ├── query.go
+│   │   └── tx.go
+│   └── rest
+│       ├── query.go
+│       └── tx.go
+├── exported
+│   └── exported.go
+├── genesis.go
```

Comment:
do we want one `genesis.go` file? we usually have the top level one just defines the import/export while the one located in `types` defines the `GenesisState` for the module.

Reply:
Yes, just one file. Types can be in types. Do you think this not clear or is confusing?

Reply:
Updated.